### PR TITLE
M1453: Review of notification to buffer messages in ScriptTask

### PR DESCRIPTION
**Pull request by srm09**
Do not merge.!

### CODE REVIEW

**Comment by muchhalsagar88** [comment]
@jdm I have addressed the review comments. Please review.

File: components/script/page.rs [hunk 1]
```diff
@@ -95,6 +95,8 @@ pub struct Page {
     /// An enlarged rectangle around the page contents visible in the viewport, used
     /// to prevent creating display list items for content that is far away from the viewport.
     pub page_clip_rect: Cell<Rect<Au>>,
+
+    pub send_notifications: Cell<bool>,
```
Comment (jdm) [inline components/script/page.rs]: Let's call this `devtools_want_updates` instead, and document it with something like "Whether the developer tools have requested live updates of page changes."

File: components/devtools_traits/lib.rs [hunk 2]
```diff
@@ -90,6 +90,8 @@ pub enum DevtoolScriptControlMsg {
     GetChildren(PipelineId, String, Sender<Vec<NodeInfo>>),
     GetLayout(PipelineId, String, Sender<(f32, f32)>),
     ModifyAttribute(PipelineId, String, Vec<Modification>),
+    RequestLiveNotifications(PipelineId),
+    StopLiveNotifications(PipelineId),
```
Comment (jdm) [inline components/devtools_traits/lib.rs]: These can be one message - WantsLiveNotification(PipelineId, bool).

File: components/script/script_task.rs [hunk 3]
```diff
@@ -626,6 +631,23 @@ impl ScriptTask {
...
             ModifyAttribute(id, node_id, modifications) =>
                 devtools::handle_modify_attribute(&*self.page.borrow(), id, node_id, modifications),
+            RequestLiveNotifications(pipeline_id) =>
+                self.handle_notifications(pipeline_id, NotificationControl::StartNotifications),
+            StopLiveNotifications(pipeline_id) =>
+                self.handle_notifications(pipeline_id, NotificationControl::StopNotifications),
+        }
+    }
+
+    fn handle_notifications(&self, pipeline_id: PipelineId, notification_type: NotificationControl) {
```
Comment (jdm) [inline components/script/script_task.rs]: This should be function in the devtools.rs file.

**Comment by jdm** [comment]
Critic stopped working because of the various merges (presumably), so I've left some comments here.

**Comment by muchhalsagar88** [comment]
I have addressed the latest comments. These changes have been delayed for a pretty long time, hope this does it..!!

**Comment by muchhalsagar88** [comment]
@jdm Does this change has to wait for the Rust version upgrade??

**Comment by jdm** [comment]
Yes; we're not merging anything non-critical until the upgrade's finished, but that should be quite soon.

**Comment by muchhalsagar88** [comment]
I am embarrassed after the last review comment..!! :(

**Comment by jdm** [comment]
Rebased in #4852.

**Pull request closed without merging**